### PR TITLE
Add support for domain and https for torrest service

### PR DESCRIPTION
**Pull request by Sam-Max**

### CODE REVIEW

**Comment by i96751414** [comment]
Thank you. I will review this as soon as I have some time.

File: lib/torrest/api.py [hunk 1]
```diff
@@ -82,7 +83,11 @@ class TorrestError(Exception):
 
 class Torrest(object):
```
Comment (i96751414) [inline lib/torrest/api.py:84]: This class/file must be agnostic to any kodi settings. I suggest passing ssl_enabled as an input for `__init__` and if true, the Torrest api would use https protocol. Also, the port should still be configurable (like it was before). One can hava https over a different port than 443.

Comment (i96751414) [reply lib/torrest/api.py:84]: Example:
```python
def __init__(self, host, port, ssl_enabled=False, session=None):
        self._base_url = "{}://{}:{}".format("https" if ssl_enabled else "http", host, port)
        self._session = session or requests
```

File: resources/settings.xml [hunk 2]
```diff
@@ -4,10 +4,10 @@
         <setting id="has_libtorrest" type="bool" default="false" visible="false"/>
         <setting id="service_enabled" type="bool" label="30059" default="true"/>
         <setting id="timeout" type="number" label="30002" default="20" enable="eq(-1,true)" subsetting="true"/>
-        <setting id="force_torrest" type="bool" label="30085" default="false" subsetting="true"
-                 enable="eq(-2,true)" visible="eq(-3,true)"/>
-        <setting id="service_ip" type="ipaddress" label="30060" default="127.0.0.1" enable="eq(-3,false)"/>
-        <setting id="port" type="number" label="30001" default="61235"/>
+        <setting id="force_torrest" type="bool" label="30085" default="false" subsetting="true" enable="eq(-2,true)" visible="eq(-3,true)"/>
+        <setting id="service_address" type="text" label="30060" default="127.0.0.1" enable="eq(-3,false)"/>
+        <setting id="port" type="number" label="30001" enable="eq(1,false)" default="61235"/>
```
Comment (i96751414) [inline resources/settings.xml:9]: This should be always enabled.

**Review by i96751414 — CHANGES_REQUESTED** [review]

File: lib/torrest/api.py (hunk 1 above)
Comment (Sam-Max) [reply lib/torrest/api.py:84]: Yes, you are right regarding the class that should be kodi agnostic and the port for https that can vary. I will update as per your instructions.

File: lib/torrest/api.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 from collections import namedtuple
-
 import requests
+from lib.settings import ssl_enabled
```
Comment (i96751414) [inline lib/torrest/api.py:3]: This import is now unused. Please revert

**Review by i96751414 — CHANGES_REQUESTED** [review]

File: lib/navigation.py [hunk 4]
```diff
@@ -13,14 +13,14 @@
     close_busy_dialog
 from lib.kodi_formats import is_music, is_picture, is_video, is_text
 from lib.player import TorrestPlayer
-from lib.settings import get_service_ip, get_port, get_buffering_timeout, show_status_overlay, get_min_candidate_size, \
-    ask_to_delete_torrent, download_after_insert, get_files_order, get_metadata_timeout
+from lib.settings import get_service_address, get_port, get_buffering_timeout, show_status_overlay, get_min_candidate_size, \
```
Comment (i96751414) [inline lib/navigation.py:16]: The linter is now complaining about this line (125 > 120 characters):
https://github.com/i96751414/plugin.video.torrest/actions/runs/5885372166/job/15961781793?pr=28

Can you fix please?

**Review by i96751414 — CHANGES_REQUESTED** [review]

File: lib/navigation.py (hunk 4 above)
Comment (Sam-Max) [reply lib/navigation.py]: Done

**Pull request merged**